shiny/ui/_input_select.py

```diff
@@ -282,6 +340,17 @@ def _normalize_choices(x: SelectChoicesArg) -> _SelectChoices:
         return x
 
 
+def _contains_html(x: _SelectChoices) -> bool:
+    for v in x.values():
+        if isinstance(v, Mapping):
+            if _contains_html(cast(_OptGrpChoices, v)):
```

Comment:
At this point, `v` is the `_Choices` values of `_OptGrpChoices` Mapping.

```suggestion
            if _contains_html(cast(_Choices), v)):
```

Reply:
With your other suggestion, this one is not relevant anymore, but you are right, that should have been `_Choices`!